drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1058,6 +1065,8 @@ static int eth_initialize(const struct device *dev)
 	return 0;
 }
 
+SYS_INIT(hal_init, POST_KERNEL, 89);
```

Comment:
The priority vale 89 shall come from KConfig. I am also somehow confused, why `SYS_INIT` statement is in this line. From code-readability perspective - it would be difficult to find. Maybe putting it just right in the end of this .c file would be better?

Reply:
I was suggested of the doc: https://docs.zephyrproject.org/apidoc/latest/group__sys__init.html which says, that we should avoid symbolic expression as the parameter. But to make it more human readable, I've add the preprocessor definition:
`#define ETH_INIT_SPLITTED 89`

Reply:
> I was suggested of the doc: https://docs.zephyrproject.org/apidoc/latest/group__sys__init.html which says, that we should avoid symbolic expression as the parameter. But to make it more human readable, I've add the preprocessor definition:
> `#define ETH_INIT_SPLIT 89`

@robi251 here is a citation from like you provided: 

> Initialization priority within _level. Note that it must be a decimal integer literal without leading zeroes or sign (e.g. 32), or an equivalent symbolic name (e.g. #define MY_INIT_PRIO 32); symbolic expressions are not permitted (e.g. CONFIG_KERNEL_INIT_PRIORITY_DEFAULT + 5).

Symbolic name is very useful, since it gives a name to develop. And by doing so it tells for what it is used for.

BTW: define is not the optimal solution. KConfig shall be better choosen.

Reply:
@robi251 please also move sys init macro to the end of the .c file.

Reply:
Nevertheless - moving `SYS_INIT` macro to the and of `.c` file is good idea.

Reply:
> Well, it's surprised for me, but using Kconfig deinition, like for example: 
> `config ETH_STM32_HAL_SPLIT_INIT_PRIO` generates weird linker error like:
> `/../arm-none-eabi/bin/ld.bfd: Undefined initialization levels used.`
> The same situation like with symbolic expression :| 

Please give me some time - I'll take a look on it.